### ISSUE TITLE
[Truffle] Super not passing correct value

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2927]
This issue comes up while running rubygems.

Example
```
class Test
  def self.new value 
    value = 5
    super
  end
  
  def initialize value
    puts "VALUE:#{value}"
  end
  
end
Test.new("Five")
```
Expected
```
ruby super.rb 
VALUE:5
```
Actual
```
 ~/Documents/jruby/bin/jruby -X+T super.rb 
VALUE:Five
```

**Comment by eregon** [comment]
The dreaded zsuper ... not sure why the local assignment does not assign to the frame though.

**Comment by chrisseaton** [comment]
We load the arguments from the original arguments array - looks like we need to load them from the locals they went into instead - not sure how that interacts with rests though...

**Comment by enebo** [comment]
@chrisseaton unnamed rests are also preserved to the call if I remember correctly.  You are correct though:

```ruby
def foo(a)
    a = 2
    super
end
```

super gets a = 2 and not what was passed in.  I think zsuper is a top 5 confusing feature for rubyists.

**Comment by chrisseaton** [comment]
Fixed in 9505ba8110bd6bfee76a85ab34dc999eacb31a15
